### PR TITLE
Workflow for combining member's data 

### PR DESCRIPTION
**Pull request by Kratika0907**
Workflow for combining member's data

### CODE REVIEW

File: .github/workflows/action.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        - name: install node 
+          run : |
+                  echo "install node"
+                  sudo apt update
+                  sudo apt install nodejs npm
+        - uses: actions/checkout@main
+        - name : Create a new dist folder 
+          run: |
+                    echo "create dist folder"
+                    node remove
```
Comment (ankushdharkar) [inline .github/workflows/action.yml:20]: Let's have a better naming here? `remove-dist-folder`?

File: .github/workflows/action.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+                    echo 'done with folder creation'
+          working-directory: ./scripts
+        - name : Create data into dist folder 
+          run: |
+                    echo "copy data"
+                    node copy
```
Comment (ankushdharkar) [inline .github/workflows/action.yml:26]: Better name?

File: .github/workflows/action.yml [hunk 3]
```diff
@@ -0,0 +1,41 @@
+name: Combine members data 
+on : 
+    push : 
+        branches:
+            - main
+jobs: 
+    job1:
+        name : data copy into dist
```
Comment (ankushdharkar) [inline .github/workflows/action.yml:8]: Let's try and be consistent please. I see later that we are using title-case, but here everything is lower-cased

File: scripts/copy.js [hunk 4]
```diff
@@ -0,0 +1,75 @@
+const fs = require("fs");
+const util = require("util");
```
Comment (ankushdharkar) [inline scripts/copy.js:2]: Prefer single quotes in JS please

File: scripts/copy.js [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+const rawImageBaseUrl =
+  "https://raw.githubusercontent.com/Real-Dev-Squad/website-static/main/members";
+
+async function fileWalker(dir) {
+  const readMemberDir = readDir(dir);
+  const personFolderNames = await readMemberDir.then(
+    (personFolder) => personFolder
+  );
+  const userDataPromises = personFolderNames.map((person) =>
```
Comment (ankushdharkar) [inline scripts/copy.js:15]: Why mixing the nomenclature? `Person` or `User`, let's try and choose one please

File: scripts/remove.js [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+const util = require("util");
+
+const readDir = util.promisify(fs.readdir);
+const delfile = util.promisify(fs.unlink);
+
+
+
+
+async function remove(dir) {
+  let flag = fs.existsSync(dir);
```
Comment (ankushdharkar) [inline scripts/remove.js:11]: const?

File: scripts/remove.js [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+  if (!flag) {
```
Comment (ankushdharkar) [inline scripts/remove.js:12]: Do we need a flag for the next line usage only?

File: scripts/create-copy-users.js [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+  );
+  const userDataPromises = usersFolderNames.map((user) =>
+    readfile(`${dir}/${user}/data.json`)
+  );
+  const userImagePromises = usersFolderNames.map((user) =>
+    readfile(`${dir}/${user}/img.png`, "base64")
+  );
+
+  const dataMap = new Map();
+  let dataPromise = Promise.all(userDataPromises).then((usersData) => {
```
Comment (ankushdharkar) [inline scripts/create-copy-users.js:23]: Why 'let'?

File: scripts/create-copy-users.js [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+          ...existData,
+        };
+        dataMap.set(user, newData);
+      } else {
+        dataMap.set(user, userData);
+      }
+    });
+  });
+
+  let imagesPromise = Promise.all(userImagePromises).then((usersImage) => {
```
Comment (ankushdharkar) [inline scripts/create-copy-users.js:41]: Re: let

File: scripts/create-copy-users.js [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+  const fullUsersData = await Promise.all([dataPromise, imagesPromise]).then(() => {
+    return Array.from(dataMap);
+  });
+  return Promise.resolve(fullUsersData);
+}
+
+fileWalker('../members').then((data) => {
+  const newData = {};
+  data.forEach((userArray) => {
+    newData[userArray[0]] = userArray[1];
```
Comment (ankushdharkar) [inline scripts/create-copy-users.js:70]: What is 0, what is 1?

File: scripts/check-remove-dist-folder.js [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+const util = require('util');
+
+const readDir = util.promisify(fs.readdir);
+const delfile = util.promisify(fs.unlink);
+
+
+
+async function remove(dir) {
+  const flag = fs.existsSync(dir);
+  if (!flag) {
```
Comment (ankushdharkar) [inline scripts/check-remove-dist-folder.js:11]: Unanswered: https://github.com/Real-Dev-Squad/website-static/pull/30#discussion_r465239164

**Comment by swarajpure** [comment]
PR no longer valid as confirmed by the author [here](https://discord.com/channels/673083527624916993/735063616168001536/797708237142884352)

**Pull request closed without merging**